xtable-core/src/main/java/org/apache/xtable/paimon/PaimonDataFileExtractor.java

```diff
@@ -78,10 +93,154 @@ private String toFullPhysicalPath(FileStoreTable table, ManifestEntry entry) {
+    BinaryArray nullCounts = stats.nullCounts();
+
+    //    log.info("Extracting stats for columns: {}", colNames);
+    //    log.info("minValues: arity={}, {}", minValues.getFieldCount(), minValues);
+    //    log.info("maxValues: arity={}, {}", maxValues.getFieldCount(), maxValues);
+    //    log.info("fieldMap: {}", fieldMap.toString());
+
+    for (int i = 0; i < colNames.size(); i++) {
+      String colName = colNames.get(i);
+      InternalField field = fieldMap.get(colName);
+      if (field == null) {
+        continue;
+      }
+
+      // Check if we already have stats for this field
+      boolean alreadyExists =
+          columnStats.stream().anyMatch(cs -> cs.getField().getPath().equals(colName));
+      if (alreadyExists) {
+        continue;
+      }
+
+      InternalType type = field.getSchema().getDataType();
+      Object min = getValue(minValues, i, type, field.getSchema());
+      Object max = getValue(maxValues, i, type, field.getSchema());
+      Long nullCount = (nullCounts != null && i < nullCounts.size()) ? nullCounts.getLong(i) : 0L;
+
+      //      log.info(
+      //          "Column: {}, Index: {}, Min: {}, Max: {}, NullCount: {}",
+      //          colName,
+      //          i,
+      //          min,
+      //          max,
+      //          nullCount);
+
+      columnStats.add(
+          ColumnStat.builder()
+              .field(field)
+              .range(Range.vector(min, max))
+              .numNulls(nullCount)
+              .numValues(rowCount)
+              .build());
+    }
+  }
+
+  private Object getValue(BinaryRow row, int index, InternalType type, InternalSchema fieldSchema) {
+    if (row.isNullAt(index)) {
+      return null;
+    }
+    switch (type) {
+      case BOOLEAN:
+        return row.getBoolean(index);
+      case INT:
+      case DATE:
+        return row.getInt(index);
+      case LONG:
+        return row.getLong(index);
+      case TIMESTAMP:
+      case TIMESTAMP_NTZ:
+        int tsPrecision;
+        InternalSchema.MetadataValue tsPrecisionEnum =
+            (InternalSchema.MetadataValue)
+                fieldSchema.getMetadata().get(InternalSchema.MetadataKey.TIMESTAMP_PRECISION);
+        if (tsPrecisionEnum == InternalSchema.MetadataValue.MILLIS) {
+          tsPrecision = 3;
+        } else if (tsPrecisionEnum == InternalSchema.MetadataValue.MICROS) {
+          tsPrecision = 6;
+        } else if (tsPrecisionEnum == InternalSchema.MetadataValue.NANOS) {
+          tsPrecision = 9;
+        } else {
+          log.warn(
+              "Field idx={}, name={} does not have MetadataKey.TIMESTAMP_PRECISION set, defaulting to default precision",
+              index,
+              fieldSchema.getName());
+          tsPrecision = TimestampType.DEFAULT_PRECISION;
+        }
+        // TODO: BinaryRow.getTimestamp().toInstant() is deprecated (use LocalZoneTimestamp), but BinaryRow does not have a method to get LocalZoneTimestamp?
```

Comment:
Q: to Paimon experts - any suggestions on getting LocalZoneTimestamp from BinaryRow?